pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +120,215 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+				if errors.IsAlreadyExists(err) {
+					return reconcile.Result{
+						RequeueAfter: 1 * time.Hour,
+					}, err
+				}
+
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// provisionMasterUserRecord does the work of provisioning the MasterUserRecord
+func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1alpha1.UserSignup, targetCluster string, logger logr.Logger) error {
+	userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+		{
+			TargetCluster: targetCluster,
+			Spec: toolchainv1alpha1.UserAccountSpec{
+				NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+					Namespaces: []toolchainv1alpha1.Namespace{},
+				},
+			},
+		},
+	}
+
+	// TODO Update the MasterUserRecord with NSTemplateTier values
+	// SEE https://jira.coreos.com/browse/CRT-74
+
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userSignup.Spec.UserID,
+			Namespace: userSignup.Namespace,
+		},
+		Spec: toolchainv1alpha1.MasterUserRecordSpec{
+			UserID: userSignup.Spec.UserID,
+			UserAccounts: userAccounts,
+		},
+	}
+
+	err := controllerutil.SetControllerReference(mur, userSignup, r.scheme)
+	if err != nil {
+		logger.Error(err, "Error setting controller reference for MasterUserRecord %s", mur.Name)
+
+		statusError := r.setStatusFailedToCreateMUR(userSignup, "failed to set controller reference")
+		if statusError != nil {
+			logger.Error(statusError, "Error setting MUR failed status")
+			return statusError
+		}
+		return err
+	}
+
+	err = r.client.Create(context.TODO(), mur)
+	if err != nil {
+		logger.Error(err, "Error creating MasterUserRecord", "UserID", userSignup.Spec.UserID)
+		if errors.IsAlreadyExists(err) {
+			statusError := r.setStatusMURAlreadyExists(userSignup, "MasterUserRecord already exists")
+			if statusError != nil {
+				logger.Error(statusError, "Error setting MUR failed status")
+				return statusError
+			}
+		} else {
+			statusError := r.setStatusFailedToCreateMUR(userSignup, "Failed to create MasterUserRecord")
+			if statusError != nil {
+				logger.Error(statusError, "Error setting MUR failed status")
+				return statusError
+			}
+		}
+
+		// Log the error
+		logger.Error(err, "Error creating MasterUserRecord", "UserID", userSignup.Spec.UserID, "TargetCluster", targetCluster)
+		return err
+	}
+	logger.Info("Created MasterUserRecord", "UserID", userSignup.Spec.UserID, "TargetCluster", targetCluster)
+
+	return err
+}
+
+// ReadUserApprovalPolicyConfig reads the ConfigMap for the toolchain configuration in the operator namespace, and returns
+// the config map value for the user approval policy (which will either be "manual" or "automatic")
+func (r *ReconcileUserSignup) ReadUserApprovalPolicyConfig() (string, error) {
+	cm, err := r.clientset.CoreV1().ConfigMaps(config.GetOperatorNamespace()).Get(config.ToolchainConfigMapName, metav1.GetOptions{})
```

Comment:
I guess, that also in this case you can use the namespace from the request so you don't need the `config.GetOperatorNamespace()` at all

Reply:
I prefer to have this controlled by the configuration, also to keep things consistent as we have a comparable function in the [member operator](https://github.com/codeready-toolchain/member-operator/blob/master/pkg/config/env.go#L7).

Reply:
But that one in member operator is used for the controller that watches cluster scoped objects (`User` and `Identity`) - in that case, the request doesn't contain the namespace. However, in your case, you watch and work only with the namespace scoped objects

Reply:
No problem, I've modified the function to accept a `namespace` parameter, and removed `env.go` altogether as it doesn't seem to be necessary any longer.